keps/sig-scheduling/5027-dra-admin-controlled-device-attributes/README.md

```diff
@@ -0,0 +1,848 @@
+should be rare.
+
+## Design Details
+
+### API
+
+The ResourceSliceOverride is a cluster-scoped type in the `resource.k8s.io` API
+group, initially in `v1alpha3` (the alpha version in Kubernetes 1.32). Because
+it may be useful to clean up after disabling the feature and because the
+device taint feature also uses this type, it gets served unconditionally as long as
+the `v1alpha3` version is enabled. Fields related specifically to this KEP
+are feature-gated.
+
+```Go
+type ResourceSliceOverride struct {
+metav1.TypeMeta
+    // Standard object metadata
+    // +optional
+    metav1.ObjectMeta
+
+    // Changing the spec automatically increments the metadata.generation number.
+    Spec ResourceSliceSpec
+}
+
+type ResourceSliceOverrideSpec struct {
+    // Devices defines how to override device attributes and taints.
+    Devices DeviceOverride
+}
+
+// DeviceOverride selects one or more devices by class, driver, pool, device names
+// and/or CEL selectors. All of these criteria must be satisfied by a device, otherwise
+// it is ignored by the override. A DeviceOverride with no selection criteria is
+// valid and matches all devices.
+type DeviceOverride struct {
+    // If DeviceClass is set, the selectors defined there must be
+    // satisfied by a device to be overridden.
+    //
+    // +optional
+    DeviceClass *string
+
+    // If driver is set, only devices from that driver are selected.
+    //
+    // +optional
+    Driver *string
+
+    // If pool is set, only devices in that pool are selected.
+    //
+    // Also setting the driver name may be useful to avoid
+    // ambiguity when different drivers use the same pool name,
+    // but this is not required because selecting pools from
+    // different drivers may also be useful, for example when
+    // drivers with node-local devices use the node name as
+    // their pool name.
+    //
+    // +optional
+    Pool *string
+
+    // If device is set, only devices with that name are selected.
+    //
+    // Also setting driver and pool may be required to avoid ambiguity,
+    // but is not required.
+    //
+    // +optional
+    Device *string
+
+    // Selectors define criteria which must be satisfied by a
+    // device to be overridden. All selectors must be satisfied.
+    //
+    // +optional
+    // +listType=atomic
+    Selectors []DeviceSelector
+
+    // ^^^
+    //
+    // Selectors is a list for the same reason why a request has a list: at some point
+    // we might have one entry which uses CEL while another uses a different, yet to be
+    // defined mechanism.
+
+    // If a ResourceSlice and a DeviceOverride define the same attribute or
+    // capacity, the value of the DeviceOverride is used. If multiple
+    // different DeviceOverrides match the same device, then the one with
+    // the highest rank wins. If the ranks are the same, it is non-deterministic
+    // which override is used.
+    Rank int
```

Comment:
A vote for `Priority` (instead of `Rank`), as I think that's the more idiomatic term for real-time collision reconciliation (see: systemd).

Reply:
Ack.

Earlier, I used "priority" for devices and wanted some other term here, but that "priority" attribute is gone now. Perhaps using "priority" for both would have been fine, too.